### PR TITLE
[iOS][Feature] Add pitch algorithm

### LINKED ISSUE #1014: Rate does not accept all values iOS
I was trying to set rate to values like 0.1845 and noticed that it does not work as intended. Instead AVPlayer seems to round the rate to the closest on of these numbers: 0.50, 0.67, 0.80, 1.0, 1.25, 1.50, and 2.0.

I think this is because the default audio time pitch algorithm only allows these values. I wonder if one of the others allows float values like this? Basically in order to fix this issue on iOS the algorithm needs to be able to be replaced. Haven't tested this on Android.

Related issue on StackOverflow: https://stackoverflow.com/questions/6630356/avplayer-rate-property-does-not-work

AVPlayerItem algorithm documentation:
https://developer.apple.com/documentation/avfoundation/avplayeritem/1385855-audiotimepitchalgorithm

Available Algorithms: https://developer.apple.com/documentation/avfoundation/avaudiotimepitchalgorithm

### PR DESCRIPTION
**Pull request by kesha-antonov**
Hey!
I added support of pitch algorithm: https://developer.apple.com/documentation/avfoundation/avaudiotimepitchalgorithm

Fixes https://github.com/react-native-community/react-native-video/issues/1014

Previously speed rate snapped to 0.50, 0.67, 0.80, 1.0, 1.25, 1.50, and 2.0.
Now we can use any speed rate.

### CODE REVIEW

File: ios/Video/RCTVideo.m [hunk 1]
```diff
@@ -967,7 +991,8 @@ - (void)applyModifiers
     [_player setVolume:_volume];
     [_player setMuted:NO];
   }
-  
+
+  [self setPitchAlgorithm:_pitchAlgorithm];
```
Comment (kesha-antonov) [inline ios/Video/RCTVideo.m:1105]: @cobarx Hey! I'm not sure if we should call setPitchAlgorithm in applyModifiers or not? I'm not ObjC dev. It works with or without it if I only change pitchAlgorithm in RN View. Not sure about other situations

File: README.md [hunk 2]
```diff
@@ -565,6 +566,18 @@ Platforms: all
 
 Note: For Android MediaPlayer, rate is only supported on Android 6.0 and higher devices.
 
+#### pitchAlgorithm
+Pitch algorithm for AVPlayerItem. [Apple documentation](https://developer.apple.com/documentation/avfoundation/avaudiotimepitchalgorithm).
+
+By default on iOS pitch algorithm is **lowQualityZeroLatency** which has limitations in allowed playback speed.
+
+* **lowQualityZeroLatency** - A low quality and very low computationally intensive pitch algorithm. This algorithm is suitable for brief fast-forward and rewind effects as well as low quality voice. The rate is snapped to {0.5, 0.666667, 0.8, 1.0, 1.25, 1.5, 2.0}.
```
Comment (cobarx) [inline README.md]: Add `(default)` next to this

File: ios/Video/RCTVideo.m (hunk 1 above)
Comment (cobarx) [reply ios/Video/RCTVideo.m:1105]: I'm not sure on this... it has been awhile and I can't remember why we have to re-apply the modifiers. Short answer is as long as the call to set it is near instantaneous it won't hurt anything.

**Review by cobarx — APPROVED** [review]
Very nicely documented. Please note the doc comment I made and update that. If you are able to verify that setting the pitch is fast.

**Comment by kesha-antonov** [comment]
Thanks!
I will do that on the next week

File: README.md (hunk 2 above)
Comment (ashnfb) [reply README.md]: @kesha-antonov can you update the docs for this and then we can approve your PR

Comment (kesha-antonov) [reply README.md]: @ashnfb @cobarx done

Comment (kesha-antonov) [reply README.md]: @cobarx @ashnfb Hey guys. Any news on this PR?

**Review by ashnfb — APPROVED** [review]

**Comment by sampsonjoliver** [comment]
What is blocking this from being merged?

**Comment by Kweiza** [comment]
What is problem with merging this PR?

**Comment by tmaly1980** [comment]
@kesha-antonov Anything you can do here to fix?

**Comment by kesha-antonov** [comment]
@tmaly1980 fix what?

**Comment by tmaly1980** [comment]
@kesha-antonov The merge conflict. This PR never was merged. I'd love to use this feature but would prefer to not work from a fork. Your changes are for v4 and master is on v5, so I'd imagine nobody is going to approve/merge the PR it until that's updated.

**Comment by kesha-antonov** [comment]
@tmaly1980 Aaahh
Didn't notice that
Yeah, I'll do that

**Comment by tmaly1980** [comment]
@kesha-antonov Any idea if this can also be implemented on Android?

**Comment by kesha-antonov** [comment]
@tmaly1980 I'm not sure. Didn't explore that

**Comment by sbycrosz** [comment]
Hello, can we merge this? I can confirm that this fork will fix the echoes when playback speed is < 1

**Comment by Bickson** [comment]
Hi, Im running this branch in a project and would love if it got merged :)

**Comment by kesha-antonov** [comment]
I'll merge recent master

**Comment by Bickson** [comment]
Anything stopping this from being merged?

**Comment by Bickson** [comment]
@kesha-antonov there is a slight pause when a algorithm I set and the volume changes, this would cause total silence when, for example, fading a song in by changing the volume. Do you know the cause of this and if it is possible to correct?
btw still think this should be merged :)

**Comment by kesha-antonov** [comment]
> @kesha-antonov there is a slight pause when a algorithm I set and the volume changes, this would cause total silence when, for example, fading a song in by changing the volume. Do you know the cause of this and if it is possible to correct?
> 
> btw still think this should be merged :)

Hi
Hmm... no, never experienced that

**Comment by kesha-antonov** [comment]
@cobarx hey
Anything else I could do to get it merged?

**Pull request closed without merging**